src/nvim/api/ui.c

```diff
@@ -706,19 +706,21 @@ void redraw(uint64_t channel_id, Array uidata, Error *error)
   Array call = ARRAY_DICT_INIT;
   char *method_name;
   size_t size;
+  Error err = ERROR_INIT; 
 
   Array uidata_copy = copy_array(uidata);
+  error = &err;
 
   for (size_t i = 0; i < uidata_copy.size; i++) {
     call = uidata_copy.items[i].data.array;
     method_name = call.items[0].data.string.data;
     size = call.items[0].data.string.size;
-    Error err = ERROR_INIT;
+    Error tmp_err = ERROR_INIT;
 
-    ApiRedrawWrapper handler_method = get_redraw_event_handler(method_name, size, &err);
-    if (ERROR_SET(&err)) {
+    ApiRedrawWrapper handler_method = get_redraw_event_handler(method_name, size, &tmp_err);
+    if (ERROR_SET(&tmp_err)) {
       logmsg(ERROR_LOG_LEVEL, "RPC: ", NULL, -1, true, "No redraw handler by name: %s", method_name);
-      error = &err;
+      err = tmp_err;
```

Comment:
This will be a memory leak if this code is executed twice.
Variables of type `Error` should not be overwritten without being freed first with `api_clear_error()`.

Reply:
Thanks a lot for this. I didn't know that. 😃